docs/tutorials/Containers.md

```diff
@@ -0,0 +1,475 @@
+    * [Cromwell: Run-in-background](#cromwell-run-in-background)
+* [Next Steps](#next-steps)
+
+### Prerequisites
+This tutorial page relies on completing the previous tutorials:
+
+* [Five Minute Introduction](FiveMinuteIntro.md)
+* [Configuration Files](ConfigurationFiles.md)
+* Recommended: [Getting started on HPC clusters](HPCIntro.md)
+
+### Goals
+
+At the end of this tutorial, you'll become familiar with container technologies and how to configure Cromwell to use these independently, or with job schedulers.
+
+
+### Specifying Containers in your Workflow
+
+Containers are specified on a per-task level, this can be achieved in WDL by specifying a [`docker`](https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#docker) tag in the `runtime` section. For example, specifying that the following WDL script should use the container `ubuntu:latest` can be achieved by:
+
+```wdl
+task hello_world {
+    String name = "World"
+    command {
+        echo 'Hello, ${name}'
+    }
+    output {
+        File out = stdout()
+    }
+    runtime {
+        docker: 'ubuntu:latest'
+    }
+}
+
+workflow hello {
+    call hello_world
+}
+```
+
+Similarly in CWL, you can specify a [`DockerRequirement`](https://www.commonwl.org/v1.0/CommandLineTool.html#DockerRequirement) inside the requirements section:
+
+```cwl
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: echo
+inputs:
+    name:
+        type: string
+        default: "World"
+        inputBinding:
+          prefix: "Hello, "
+outputs:
+    out: stdout
+
+requirements:
+    DockerRequirement:
+        dockerPull: "ubuntu:latest"
+``` 
+___
+### Docker
+
+[Docker](https://www.docker.com) is a popular container technology that is natively supported by Cromwell and WDL.
+
+
+#### Docker on a Local Backend
+
+On a single machine (laptop or server), no extra configuration is needed to allow docker to run, provided Docker is installed.
+
+You can install Docker for Linux, Mac or Windows from [Docker Hub](https://hub.docker.com/search/?type=edition&offering=community)
+
+#### Docker on Cloud
+
+It is strongly advised that you provide a Docker image to tasks that will run on Cloud backends, and in fact most Cloud providers require it.
+
+It might be possible to use an alternative container engine, but this is not recommended if Docker is supported.
+
+#### Docker on HPC
+
+Docker can allow running users to gain superuser privileges, called the [Docker daemon attack surface](https://docs.docker.com/engine/security/security/#docker-daemon-attack-surface). In HPC and multi-user environments, Docker recommends that "only trusted users should be allowed to control your Docker Daemon".
+
+For this reason it's worth exploring other technologies that will support the reproducibility and simplicity of running a workflow that uses docker containers; Singularity or udocker.
+
```

Comment:
> Singularity or udocker, discussed next.

I think we should add that, because my first instinct was "Where is the link to learn more about this?" and I almost suggested adding it. But if the reader knows it's coming next, no need! It's coming :)

Reply:
I changed it to lead better towards the next sections, I'm pretty happy to change it back and add the "discussed next" if you don't think this change adds value.
> For this reason, this tutorial will also explore other technologies that support the reproducibility and simplicity of running a workflow that use docker containers; Singularity and udocker.